cypress/support/step_definitions/validations.js

```diff
@@ -63,6 +63,32 @@ Given(
             : CONSTANTS.NULL_RESPONSE;
           const expectingError = item.expectingError;
           const isNullCase = item.isNullCase || false;
+          // check if the dynamic device details env is present
+          if (
+            Cypress.env(CONSTANTS.DEVICE_DATA) &&
+            Object.keys(Cypress.env(CONSTANTS.DEVICE_DATA)).length > 0
+          ) {
+            let type;
+            // if the validation object used for current validation contains the required data
+            if (contentObject && contentObject.data) {
+              for (let i = 0; i < contentObject.data.length; i++) {
+                if (
+                  contentObject.data[i].validations &&
+                  contentObject.data[i].validations[0] &&
+                  contentObject.data[i].validations[0].type &&
+                  contentObject.data[i].validations[0].mode == CONSTANTS.DEVICE_CONTENT_VALIDATION
+                ) {
+                  type = contentObject.data[i].validations[0].type;
+                  // if the dynamic device details env contains the validation key
+                  if (Cypress.env(CONSTANTS.DEVICE_DATA).hasOwnProperty(type)) {
+                    contentObject.data[i].validations[0].type = Cypress.env(CONSTANTS.DEVICE_DATA)[
+                      type
+                    ];
+                  }
+                }
+              }
+            }
+          }
```

Comment:
Do we require else condition to notify about Device data not found or fetching the data from mac.json?

Reply:
Added else condition for the case were deviceData env is empty